app/javascript/utils/lazy-highlight-all.ts

```diff
@@ -0,0 +1,54 @@
+export function lazyHighlightAll(): void {
+  let highlighted = false
+
+  function loadAndHighlight() {
+    import('@/utils/highlight')
+      .then((m) => {
+        m.highlightAll()
+        observer.disconnect()
+      })
+      .catch((e) => {
+        observer.disconnect()
+        throw new Error(e)
+      })
+  }
+
+  function checkAndHighlight(): void {
+    // only load highlightjs when a code block with lang-... or language-... classname exists in the DOM
+    if (highlighted || !hasHighlightableCodeBlock(document)) return
```

Comment:
I would remove the `hasHighlightableCodeBlock` call here and move it to the calling site. This is already done in `mutationCallback`, in which case the `hasHighlightableCodeBlock` function would be called twice.



```suggestion
  function highlight(): void {
    if (highlighted) return
```